### ISSUE TITLE
How to map devices in device tree to pxt-jacdac packages?

### ISSUE THREAD

**Issue by jamesadevine** [microsoft/jacdac#797]
It's unclear how devices in the device tree map to service clients in pxt-jacdac. This is problematic if you are using mkc and visual studio code to write programs for makecode + jacdac.

**Comment by pelikhan** [comment]
Add to this bug

**Comment by jamesadevine** [comment]
> you can do `mkc add jacdac-joystick` already
> 
> can we have more details about this?

Right, that's fine if you know joystick maps to "jacdac-joystick" in pxt-jacdac. The issue is, for someone who isn't us and is using mkc + visual studio, how do they work out what packages to specify in the `mkc add` command

**Comment by jamesadevine** [comment]
It would be cool for the mkc command to auto detect modules, but i understand there are dragons with usb via node...

**Comment by jamesadevine** [comment]
sample output?

```
mkc jdls
-> rotary -> jacdac-rotary
-> button -> jacdac-button
-> joystick -> jacdac-joystick
Auto install? [y/n]
```

**Comment by pelikhan** [comment]
we should extend search to also return jacdac sub-repoes. the cloud search does not support mono-repoes

**Comment by tballmsft** [comment]
@pelikhan - does this need to be documented somewhere?

**Comment by pelikhan** [comment]
it should go here https://microsoft.github.io/jacdac-docs/clients/makecode/code/
